src/Uno.UI.Composition/Composition/KeyFrameAnimations/ScalarKeyFrameAnimation.cs

```diff
@@ -0,0 +1,50 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Uno;
+
+namespace Microsoft.UI.Composition;
+
+public partial class ScalarKeyFrameAnimation : KeyFrameAnimation
+{
+#if !__IOS__
+	private readonly SortedDictionary<float, float> _keyFrames = new();
+#endif
+
+	internal ScalarKeyFrameAnimation(Compositor compositor) : base(compositor)
+	{
+	}
+
+#if !__IOS__
+	private protected override int KeyFrameCountCore => _keyFrames.Count;
+
+	public void InsertKeyFrame(float normalizedProgressKey, float value)
+		=> _keyFrames[normalizedProgressKey] = value;
+
+	[NotImplemented]
+	public void InsertKeyFrame(float normalizedProgressKey, float value, CompositionEasingFunction easingFunction)
+		=> InsertKeyFrame(normalizedProgressKey, value);
+
+
+	internal override object Start()
+	{
+		base.Start();
+		if (!_keyFrames.TryGetValue(0, out var startValue))
+		{
+			// TODO: Set startValue to be the current property value.
```

Comment:
Missing code here also? And it seems to be the same for all `***KeyFrameAnimation`.